Testnet/MintableToken/MintableToken/MintableToken.cs

```diff
@@ -221,6 +235,28 @@ public void MintWithMetadata(Address account, UInt256 amount, string metadata)
         Log(new MintMetadata() { To = account, Amount = amount, Metadata = metadata });
     }
 
+    /// <inheritdoc />
+    public void MintWithMetadataForNetwork(Address account, UInt256 amount, string metadata, string destinationAccount, string destinationNetwork)
+    {
+        MintWithMetadata(account, amount, metadata);
+
+        if (string.IsNullOrEmpty(destinationAccount) && ((string.IsNullOrEmpty(destinationNetwork) || destinationNetwork == "CIRRUS")))
+        {
+            return;
+        }
+
+        Assert(!string.IsNullOrEmpty(destinationAccount) && !string.IsNullOrEmpty(destinationNetwork) && destinationNetwork != "CIRRUS", "Invalid destination");
```

Comment:
It looks bit weird to me see a network specific code in smart contracts. Smart contracts should not aware of network as design.